### PR TITLE
Interface atlases support + Refresh frames icons

### PR DESCRIPTION
**Pull request by Raenore**
This commit handles a few things:
- Icons in tooltips are now 32x32 instead of 25x25.
- Introduce interface atlas support.
- Refresh the frames (Target and Tooltip frames) icons with some new HD-er versions.
- Removed classic/era fallback icons that were no longer necessary.
- Target frame tooltips now also have title icons in their tooltips.
- Updated `getTooltipTitleWithIcon` usage of `Utils.str.icon` with `TRP3_MarkupUtil.GenerateIconMarkup`.

> [!TIP]
> A gist will be included in the PR to show the before/after of the icons:
https://gist.github.com/Raenore/7463cf6e9baeec4c35a1c62859e6d1fe

> [!IMPORTANT]  
> This PR has an Extended counterpart found: https://github.com/Total-RP/Total-RP-3-Extended/pull/249

### CODE REVIEW

**Review by Meorawr — CHANGES_REQUESTED** [review]
I think you've missed the point of `iconFile`. The intent was that you'd keep the existing `icon` field and _add both_ `iconFile` and `iconAtlas`, with any buttons using those fields having their values passed through to SetIconTextureToFile or SetIconTextureToAtlas.

The behaviour of `icon` would be unchanged, minus to remove the hack for the report icon (which should use `iconFile`).

**Comment by Raenore** [comment]
> I think you've missed the point of `iconFile`. The intent was that you'd keep the existing `icon` field and _add both_ `iconFile` and `iconAtlas`, with any buttons using those fields having their values passed through to SetIconTextureToFile or SetIconTextureToAtlas.
> 
> The behaviour of `icon` would be unchanged, minus to remove the hack for the report icon (which should use `iconFile`).

Should be addressed in latest commit.

File: totalRP3/Modules/TargetFrame/TargetFrame.lua [hunk 1]
```diff
@@ -99,15 +114,38 @@ local function onStart()
...
+						else
+							-- This handles the Ellyb.Icon() icons.
+							uiButton:SetIconTexture(buttonStructure.iconFile:GetFileID())
+						end
+					else
+						uiButton:SetIconTexture(buttonStructure.iconFile);
+					end
+				end
 			end
+			]]
```
Comment (Meorawr) [inline totalRP3/Modules/TargetFrame/TargetFrame.lua:148]:
```suggestion
```
yeetus

File: totalRP3/Modules/TargetFrame/TargetFrame.lua [hunk 2]
```diff
@@ -99,15 +114,38 @@ local function onStart()
...
-					uiButton:SetIconTexture(buttonStructure.icon:GetFileID())
-				end
-			else
+			-- icon > iconAtlas > iconFile so sayeth the Meorawr.
+			if buttonStructure.icon then
 				uiButton:SetIconTexture(buttonStructure.icon);
+			elseif buttonStructure.iconAtlas and C_Texture.GetAtlasInfo(buttonStructure.iconAtlas) then
+				uiButton:SetIconTextureToAtlas(buttonStructure.iconAtlas);
+			elseif buttonStructure.iconFile then
+				uiButton:SetIconTextureToFile(buttonStructure.iconFile)
```
Comment (Meorawr) [inline totalRP3/Modules/TargetFrame/TargetFrame.lua:123]:
```suggestion
				uiButton:SetIconTextureToFile(buttonStructure.iconFile);
```
Lamb sauce required.

File: totalRP3/Modules/Register/Characters/ReportProfileButton.lua [hunk 3]
```diff
@@ -31,8 +29,8 @@ TRP3_API.RegisterCallback(TRP3_Addon, TRP3_Addon.Events.WORKFLOW_ON_LOADED, func
 			end
 			Ellyb.Popups:OpenURL("https://battle.net/support/help/product/wow/197/1501/solution", reportText, nil, loc.COPY_SYSTEM_MESSAGE);
 		end,
-		tooltip =  REPORT_ICON:GenerateString(25) .. loc.REG_REPORT_PLAYER_PROFILE,
+		tooltip =  loc.REG_REPORT_PLAYER_PROFILE,
```
Comment (Meorawr) [inline totalRP3/Modules/Register/Characters/ReportProfileButton.lua:32]:
```suggestion
		tooltip = loc.REG_REPORT_PLAYER_PROFILE,
```
The spacing is fucking raw.

**Review by Meorawr — CHANGES_REQUESTED** [review]

File: totalRP3/Modules/TargetFrame/TargetFrame.lua [hunk 4]
```diff
@@ -41,6 +41,21 @@ local function onStart()
 	local marginLeft = 10;
 	local loaded = false;
 
+	--- getTooltipTitleWithIcon creates a tooltip title string with icon in front of it by its atlas or icon.
+	---@param buttonStructure table Contains all the button data.
+	---@return string # The tooltip tiple with an icon in front.
```
Comment (Solanya) [inline totalRP3/Modules/TargetFrame/TargetFrame.lua:46]:
```suggestion
	---@return string # The tooltip title with an icon in front.
```

**Review by Meorawr — APPROVED** [review]

**Pull request merged**